Windows mono release encounters error C2375: 'GetAzBudgetRHI': redefinition; different linkage

**Describe the bug**
When building the installer on the `development` branch, the build fails with a C2375 error

**Assets required**
https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE-development_nightly-installer/detail/O3DE-development_nightly-installer/326/pipeline/247

**Steps to reproduce**
Steps to reproduce the behavior:
1. Run a build using the `install_mono_release` target
2. Wait until the build processes the `RHI/MemoryView.cpp` file
3. See error

**Expected behavior**
Compile completes successfully

**Actual behavior**
The build fails with the following message:

```
D:/workspace/o3de/Gems/Atom/RHI/DX12/Code/Source/RHI/MemoryView.cpp(16,1): error C2375: 'GetAzBudgetRHI': redefinition; different linkage [D:\workspace\o3de\build\windows_mono\External\DX12-f9fb3873\Code\Atom_RHI_DX12.Private.Static.vcxproj]

AZ_DECLARE_BUDGET_SHARED(RHI);
^

D:\workspace\o3de\Gems\Atom\RHI\Code\Include\Atom/RHI.Reflect/Base.h(28,1): note: see declaration of 'GetAzBudgetRHI'

AZ_DECLARE_BUDGET(RHI);
^
```

**Screenshots/Video**
If applicable, add screenshots and/or a video to help explain your problem.

**Found in Branch**
This occurs in the `development` and `stabilization/25100` branches

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
[`bcce98f587977f7bf91b79f2fa37cb55ae1a8e1f`](https://github.com/o3de/o3de/commit/bcce98f587977f7bf91b79f2fa37cb55ae1a8e1f)

**Desktop/Device (please complete the following information):**
 - Device: AWS EC2
 - OS: Windows
 - Version: 2022 Server
 - CPU: AMD EPYC
 - GPU: N/A
 - Memory: 32GB

**Additional context**
Startup log except:

```
Building for: Visual Studio 17 2022
Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.
The C compiler identification is MSVC 19.36.32532.0
The CXX compiler identification is MSVC 19.36.32532.0

Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped

Basic Configuration Information:

CMAKE_VERSION: 3.24.0
CMAKE_GENERATOR: Visual Studio 17 2022
CMAKE_CXX_COMPILER_ID: MSVC
CMAKE_CXX_COMPILER_VERSION: 19.36.32532.0
O3DE_INSTALL_VERSION_STRING: 4.2.0
```

--- a/Gems/Atom/RHI/Code/Source/RHI/AsyncWorkQueue.cpp
+++ b/Gems/Atom/RHI/Code/Source/RHI/AsyncWorkQueue.cpp
@@ -9,7 +9,11 @@
 
 #include <AzCore/Debug/Profiler.h>
 
+#if defined(AZ_MONOLITHIC_BUILD)
+AZ_DECLARE_BUDGET(RHI);
+#else
 AZ_DECLARE_BUDGET_SHARED(RHI);
+#endif
 
 namespace AZ::RHI
 {

--- a/Gems/Atom/RHI/DX12/Code/Source/RHI/MemoryView.cpp
+++ b/Gems/Atom/RHI/DX12/Code/Source/RHI/MemoryView.cpp
@@ -13,7 +13,11 @@
 #include <AzCore/std/string/string.h>
 #include <dx12ma/D3D12MemAlloc.h>
 
+#if defined(AZ_MONOLITHIC_BUILD)
+AZ_DECLARE_BUDGET(RHI);
+#else
 AZ_DECLARE_BUDGET_SHARED(RHI);
+#endif
 
 namespace AZ
 {

